dotCMS/src/test/java/com/dotmarketing/db/DataSourceStrategyProviderTest.java

```diff
@@ -99,30 +106,84 @@ public void testGet(final String testCase)
+     * <ul>
+     *            <li>Custom provider</li>
+     *            <li>db.properties file</li>
+     *            <li>System environment variables</li>
+     *            <li>Docker secrets</li>
+     *            <li>context.xml</li>
+     * </ul>
+     *
+     * Expected result: A DataSource is initialized using the {@link TomcatDataSourceStrategy} provider
+     * @param testCase
      * @throws IllegalAccessException
      * @throws InstantiationException
      * @throws ClassNotFoundException
      */
-    @Test(expected=ClassNotFoundException.class)
-    public void testGetUsingCustomProvider()
+    @UseDataProvider("testCases")
+    @Test
+    public void testGetFallback(final String testCase)
             throws IllegalAccessException, InstantiationException, ClassNotFoundException {
 
         final DataSourceStrategyProvider provider = Mockito.spy(DataSourceStrategyProvider.class);
+        final SystemEnvironmentProperties properties = Mockito.spy(SystemEnvironmentProperties.class);
+        final DBPropertiesDataSourceStrategy dbStrategy = Mockito.spy(DBPropertiesDataSourceStrategy.class);
+        final SystemEnvDataSourceStrategy systemEnvStrategy = Mockito.spy(SystemEnvDataSourceStrategy.class);
+        final DockerSecretDataSourceStrategy dockerSecretStrategy = Mockito.spy(DockerSecretDataSourceStrategy.class);
+        final TomcatDataSourceStrategy tomcatDataSourceStrategy = Mockito.mock(TomcatDataSourceStrategy.class);
+
+        final HikariDataSource dummyDatasource = new HikariDataSource();
+
+        Mockito.when(provider.getSystemEnvironmentProperties()).thenReturn(properties);
+        Mockito.when(provider.getDBPropertiesInstance()).thenReturn(dbStrategy);
+        Mockito.when(provider.getSystemEnvDataSourceInstance()).thenReturn(systemEnvStrategy);
+        Mockito.when(provider.getDockerSecretDataSourceInstance()).thenReturn(dockerSecretStrategy);
+        Mockito.when(provider.getTomcatDataSourceInstance()).thenReturn(tomcatDataSourceStrategy);
 
-        Mockito.when(provider.getCustomDataSourceProvider()).thenReturn("DummyProvider");
+        if (testCase.equals("DBProperties")) {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(true);
+        } else {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(false);
+        }
+
+        if (testCase.equals("SystemEnv")){
+            Mockito.when(properties.getVariable("connection_db_base_url")).thenReturn("dummy_url");
+        } else {
+            Mockito.when(properties.getVariable("connection_db_base_url")).thenReturn(null);
+        }
+
+        if (testCase.equals("DockerSecret")) {
+            Mockito.when(dockerSecretStrategy.dockerSecretPathExists()).thenReturn(true);
+        } else {
+            Mockito.when(dockerSecretStrategy.dockerSecretPathExists()).thenReturn(false);
+        }
+
+        if (testCase.equals("CustomProvider")){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5092420)